contracts/AssetPool.sol

```diff
@@ -253,6 +263,94 @@ contract AssetPool is Ownable {
 
+    /// @notice Transfers collateral tokens to a new Asset Pool which previously
+    ///         was approved by the governance. New underwriter tokens will be
+    ///         received to this contract but immediately forwarded to the right
+    ///         owner - underwriter. Old underwriter tokens are burned in favor
+    ///         of new tokens minted in a new Asset Pool.
+    /// @param covAmount Amount of underwriter tokens used to calculate collateral
+    ///                  tokens which are transferred to a new asset pool.
+    function upgradeToNewAssetPool(uint256 covAmount) external {
+        /* solhint-disable not-rely-on-time */
+        require(
+            covAmount > 0,
+            "Underwriter token amount must be greater than 0"
+        );
+
+        uint256 covBalance = underwriterToken.balanceOf(msg.sender);
+        require(
+            covAmount <= covBalance,
+            "Underwriter token amount exceeds available balance"
+        );
+
+        require(
+            address(newAssetPool) != address(0),
+            "New asset pool must be assigned"
+        );
+
+        require(
+            address(newUnderwriterToken) != address(0),
+            "New underwriter token must be assigned"
+        );
+
+        uint256 covSupply = underwriterToken.totalSupply();
+
+        // slither-disable-next-line reentrancy-events
+        rewardsPool.withdraw();
+
+        uint256 collateralBalance = collateralToken.balanceOf(address(this));
+
+        uint256 collateralAmountToTransfer =
+            covAmount.mul(collateralBalance).div(covSupply);
+
+        collateralToken.safeApprove(
+            address(newAssetPool),
+            collateralAmountToTransfer
+        );
+        // collateralAmountToTransfer will be sent to a new AssetPool and new
+        // underwriter tokens will be mint to this contract
+        newAssetPool.deposit(collateralAmountToTransfer);
+
+        uint256 newUnderwriterTokenBalance =
+            newUnderwriterToken.balanceOf(address(this));
+        // underwriter tokens are immidiately transferred to the underwriter who
+        // transferred collateral to a new asset pool
+        newUnderwriterToken.safeTransfer(
+            msg.sender,
+            newUnderwriterTokenBalance
+        );
+
+        emit UpgradeToNewAssetPoolCompleted(
+            msg.sender,
+            collateralAmountToTransfer,
+            newUnderwriterTokenBalance,
+            block.timestamp
+        );
+
+        // old underwriter tokens are burned in favor of new minted in a new asset pool
+        underwriterToken.burnFrom(msg.sender, covAmount);
+    }
+
+    /// @notice Allows governance to set a new asset pool and a new underwriter
+    ///         token so the underwriters can move their collateral tokens to
+    ///         a new asset pool.
+    function approveNewAssetPoolUpgrade(
+        IAssetPool _newAssetPool,
+        UnderwriterToken _newUnderwriterToken
+    ) external onlyOwner {
```

Comment:
The owner of this contract is the coverage pool so this will not work unless the idea was to route this call through `CoveragePool` which I think is acceptable and probably the best solution here.

Reply:
Yup, that's the idea, will add this change in the next commits.